### ISSUE TITLE
CD Version 4 Tracking Issue

### ISSUE THREAD

**Issue by melotic** [microsoft/component-detection#770]
Tracking issue for version 4.0 of Component Detection. This is a list of changes to make before shipping breaking changes.

```[tasklist]
### Tasks
- [ ] #233
- [ ] #452
- [ ] Switch to .NET 8 RC
- [ ] #251
- [ ] https://github.com/microsoft/component-detection/issues/772
```

AB#2102422

**Comment by JamieMagee** [comment]
Some additional context for .NET 8 Release Candidate's support, from the [.NET and .NET Core Support Policy][1]

> Go-live releases are supported by Microsoft in production. These are typically our release candidate builds, just before the generally available (GA) release.

[1]: https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

**Comment by cobya** [comment]
Closing as v4 was already released - #1135 will handle tracking for v5.
